### DISCUSSION TITLE
Copilot content exclusions – Temporary rollback and upcoming fix

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#76344] [Copilot Conversations] [answered]
Shortly after releasing [Copilot content exclusions](https://github.blog/changelog/2023-11-08-copilot-content-exclusion-is-now-available-in-public-beta/) on November 8, 2023, our team observed that the feature was causing clients to be incorrectly blocked from using Copilot. This necessitated an immediate rollback of this feature.

### **What Happened?**  
Once the feature was enabled for all **Copilot Business customers**, we observed a spike in errors and some end-users being completely blocked from using Copilot. The problem was related to the way content exclusions policies are fetched from the client.

### **Current Actions and Next Steps**  

Our engineering team is engaged in deploying the necessary fixes. We have identified the faulty code in the client and are also deploying more verifications both server and client side to ensure this does not happen again. However, we want to approach the reintroduction of this feature with caution. Customers who had previously setup a content exclusions configuration are not affected by the rollback.

 **We expect to re-deploy the feature within the next few weeks.**

For any queries or feedback, please feel free to share them in the comments below.

**Comment by DylanRR** [comment]
As of an hour ago I was having no issues but now no matter what I ask copilot it says its blocked due to matching public code. Looks like I'm not the only one. See https://github.com/orgs/community/discussions/76507

**Reply by Akash1134** [reply]
Hi @DylanRR , 

We're sorry about the trouble caused. It seems you might be impacted by a bad deploy that happened [yesterday](https://github.com/microsoft/vscode-copilot-release/issues/583) 

Please restart VSCode and logout/login to the Copilot extension. This issue has already been resolved.

Thanks!

**Comment by Msalinas1996** [comment]
You a hoe

**Comment by Akash1134** [comment, marked as answer]
## Update on Copilot Content Exclusion 🆕  

Following our [previous](https://github.blog/changelog/2023-11-08-copilot-content-exclusion-is-now-available-in-public-beta) communication dated November 8, 2023, regarding the temporary rollback of the Copilot content exclusions feature, we are pleased to announce the re-deployment of this feature with significant enhancements. The rollout will be progressive during the next 10 days as we monitor the behaviour.

With Content Exclusion, **GitHub Copilot Business customers will be able to prevent specified files or repositories from being used to inform code completion suggestions made by GitHub Copilot**. GitHub Copilot will not be available in excluded files. Organization administrators or repository owners can choose which files or repositories are excluded. [Learn more](https://docs.github.com/en/copilot/managing-copilot-business/configuring-content-exclusions-for-github-copilot).

### Overview of the Issue 👀 

Our team observed a critical issue where clients were incorrectly blocked from using Copilot due to the initial implementation of content exclusions. This was primarily caused by errors in fetching content exclusion policies from the client, leading to a temporary suspension of the feature.

### Actions Undertaken 🏁 

In response to this, our engineering team undertook a comprehensive review and rectification process. The issues identified in the client's code were addressed, and additional verifications were implemented on both server and client sides to prevent recurrence.

### New Enhancements in the Re-deployed Feature 🆕 

*   **Performance Update**: We have optimized the performance of the content exclusions feature, ensuring minimal impact on the user experience.
*   **Extended Coverage**: The feature now supports all our official Integrated Development Environments: Visual Studio, JetBrains IDEs, Visual Studio Code, and Vim/Neovim.

### Current Status 🟢 

*   Users with pre-existing content exclusion configurations will experience no change.
*   New and returning users can now utilize the enhanced feature across all supported IDEs.

**Up Next** ❓  We are closely monitoring the performance and user feedback post-deployment. The support for Copilot Chat is also in progress and will be part of the General Availability.
